proto/cluster.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+option go_package = "github.com/ray-project/kuberay/proto/go_client";
+package proto;
+
+import "google/protobuf/timestamp.proto";
+
+
+message Cluster {
+  // Required input field. Unique Cluster name provided by user.
+  string name = 1;
+
+  // Required input field. Cluster's namespace provided by user
+  string namespace = 2;
+
+  // Required field. This field indicates the user who owns the cluster.
+  string user = 3;
+
+  // Optional input field. Ray cluster version
+  string version = 4;
+
+  // Optional field.
+  enum Environment {
+    DEV = 0;
+    TESTING = 1;
+    STAGING = 2;
+    PRODUCTION = 3;
+  }
+  Environment environment = 5;
+
+  // Required field. This field will be used to retrieve right ray container
+  string image = 6;
+  
+  // Required field. This field indicates ray cluster configuration
+  ClusterSpec cluster_spec = 7;
+
+  // Output. The time that the Cluster created.
+  google.protobuf.Timestamp created_at = 8;
+
+  // Output. The time that the Cluster deleted.
+  google.protobuf.Timestamp deleted_at = 9;
+}
+
+message ClusterSpec {
+  // The name of the cluster spec
+  string name = 1;
+  // The head group configuration
+  HeadGroupSpec head_group_spec = 2;
+  // The worker group configurations
+  repeated WorkerGroupSpec worker_group_sepc = 3;
+}
+
+message HeadGroupSpec {
+  // Optional. The computeTemplate of head node group
+  ComputeTemplate resource = 1;
+  // Optional. The ray start parames of head node group
+  map<string, string> ray_start_params = 2;
+  // Optional. The service type (ClusterIP, NodePort, Load balancer) of the head node
+  string service_type = 3;
+  // Output: internal/external service endpoint
+  string service_address = 4;
+}
+
+message WorkerGroupSpec {
```

Comment:
How about add image in `WorkerGroupSpec`? We can specify different image for WorkerGroupSpec

Reply:
That's reasonable and I will do it.

Reply:
Done. Also add it inside the head.